node/src/archive/transformer.rs

```diff
@@ -0,0 +1,158 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use std::collections::BTreeMap;
+
+use execution_core::signatures::bls::PublicKey as AccountPublicKey;
+use execution_core::transfer::withdraw::WithdrawReceiver;
+use execution_core::transfer::{
+    ConvertEvent, MoonlightTransactionEvent, WithdrawEvent, CONVERT_TOPIC,
+    MINT_TOPIC, MOONLIGHT_TOPIC, TRANSFER_CONTRACT, WITHDRAW_TOPIC,
+};
+use node_data::events::contract::{ContractEvent, ContractTxEvent, TxHash};
+use serde::{Deserialize, Serialize};
+
+/// More efficient format for events that belong to the same tx to not duplicate
+/// TxHash
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct MoonlightTxEvents {
+    events: Vec<ContractEvent>,
+    origin: TxHash,
+}
+
+impl MoonlightTxEvents {
+    // Private on purpose
+    fn new(events: Vec<ContractEvent>, origin: TxHash) -> Self {
+        Self { events, origin }
+    }
+
+    pub fn events(&self) -> &Vec<ContractEvent> {
+        &self.events
+    }
+
+    pub fn origin(&self) -> &TxHash {
+        &self.origin
+    }
+}
+
+type AddressMapping = (AccountPublicKey, TxHash);
+type MemoMapping = (Vec<u8>, TxHash);
+
+/// Groups the events from a block by their origin and returns
+/// only the groups that contain a moonlight in- or outflow
+///
+/// Returns the address mappings, memo mappings and groups
+pub(super) fn group_by_origins_filter_and_convert(
+    block_events: Vec<ContractTxEvent>,
+) -> (
+    Vec<AddressMapping>,
+    Vec<MemoMapping>,
+    Vec<MoonlightTxEvents>,
+) {
+    // 1st Group events by origin (TxHash) & throw away the ones that
+    // don't have an origin
+    let mut moonlight_is_already_grouped: BTreeMap<TxHash, Vec<ContractEvent>> =
+        BTreeMap::new();
+    for event in block_events {
+        if let Some(origin) = event.origin {
+            let event_to_analyze = event.event;
+
+            moonlight_is_already_grouped
+                .entry(origin)
+                .or_default()
+                .push(event_to_analyze);
+        } else {
+            continue;
```

Comment:
Why is the "continue" needed here? If else part is removed, the meaning is the same unless I miss sth.